source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -215,8 +215,15 @@ int DefaultCertValidator::doVerifyCertChain(
     }
   }
 
-  return allow_untrusted_certificate_ ? 1
-                                      : (validated != Envoy::Ssl::ClientValidationStatus::Failed);
+  // If `trusted_ca` exists, it is already verified in the code above. Thus, we just need to make
+  // sure the verification for other validation context configurations doesn't fail (i.e. either
+  // `NotValidated` or `Validated`). If `trusted_ca` doesn't exist, we will need to make sure other
+  // configurations are verified and the verification succeed.
+  int validation_status = verify_trusted_ca_
```

Comment:
`bool`?

Reply:
Thanks for review! I think either bool or int will work here as bool <-> int implicit conversation will be triggered.
The reason I used int here is to align with value 1 in ternary operator in line 226 and the return type of this function (which is int).

Please let me know what do you think.

Reply:
I think `int` is fine given that the function returns the same error code as `X509_verify_cert`